crates/wadm/src/scaler/daemonscaler/provider.rs

```diff
@@ -81,10 +85,12 @@ impl<S: ReadStore + Send + Sync + Clone> Scaler for ProviderDaemonScaler<S> {
         match event {
             Event::ProviderStarted(ProviderStarted { provider_id, .. })
             | Event::ProviderStopped(ProviderStopped { provider_id, .. })
-                if provider_id == &self.config.provider_id =>
-            {
-                self.reconcile().await
-            }
+            | Event::ProviderHealthCheckFailed(ProviderHealthCheckFailed {
+                data: ProviderHealthCheckInfo { provider_id, .. },
+            })
+            | Event::ProviderHealthCheckPassed(ProviderHealthCheckPassed {
+                data: ProviderHealthCheckInfo { provider_id, .. },
```

Comment:
In this case, I actually think that we should update the status in the case of a failed health check but not run a full reconcile. The main reason being that for a failed health check we don't really want to try and reconcile with commands, but we do want to update the status to be Failed. Does that make sense?

Reply:
Your other modifications in `reconcile()` look good though, and I think that would be able to show the failed status after we set it in response to that event. Here (if I remember right) there's also a message on the event that you can set the status to

Reply:
Yep, makes sense! I'll make the change to set the status without running a full reconcile.

Reply:
The health check event doesn't have any message attached to it so I haven't made any changes to the status message.